crypto/evp/p_lib.c

```diff
@@ -535,6 +535,38 @@ int EVP_PKEY_assign(EVP_PKEY *pkey, int type, void *key)
     return (key != NULL);
 }
 
+int EVP_PKEY_deassign(EVP_PKEY *pkey)
+{
+    if (pkey == NULL)
+        return 1;
+
+    /* If it was downgraded, we refuse to make it provider only again */
+    if (pkey->downgraded)
+        return 1;
+
+    /* Make sure it's exported to a provider */
+    {
+        void *keydata = NULL;
```

Comment:
Does this need to be in its own block?  The function isn't large and the variables can be declared at the top just as easily.

Reply:
Do I have to?  Also, experimental, I really don't expect this code to end up in the main source, so really, [shrug]